index.js

```diff
@@ -68,6 +68,22 @@ function formatWithCursor(text, opts, addAlignmentSize) {
   }
 
   text = stripBom(text);
+
+  if (
+    opts.insertPragma &&
+    !hasPragma(text) &&
+    opts.rangeStart === 0 &&
+    opts.rangeEnd === Infinity
+  ) {
+    const parsedDocblock = docblock.parseWithComments(docblock.extract(text));
+    const pragmas = Object.assign({}, { format: "" }, parsedDocblock.pragmas);
+    const newDocblock = docblock.print({
+      pragmas,
+      comments: parsedDocblock.comments.trim()
```

Comment:
I actually removed the `.trim()` here: facebook/jest#4576

Reply:
Take a look at [this snapshot](https://github.com/prettier/prettier/pull/2865/files#diff-f83be32c20a86f62fb25697b07ccbd20R76).  

It seems to me like we should remove leading newlines but preserve leading whitespace